pex/venv/installer.py

```diff
@@ -17,17 +18,119 @@
 from pex.environment import PEXEnvironment
 from pex.orderedset import OrderedSet
 from pex.pep_376 import InstalledWheel, LoadError
+from pex.pep_440 import Version
+from pex.pep_503 import ProjectName
 from pex.pex import PEX
+from pex.result import Error
 from pex.tracer import TRACER
 from pex.typing import TYPE_CHECKING
 from pex.util import CacheHelper
 from pex.venv.bin_path import BinPath
 from pex.venv.install_scope import InstallScope
-from pex.venv.virtualenv import Virtualenv
+from pex.venv.virtualenv import PipUnavailableError, Virtualenv
 
 if TYPE_CHECKING:
     import typing
-    from typing import Iterable, Iterator, Optional, Tuple
+    from typing import DefaultDict, Iterable, Iterator, List, Optional, Tuple, Union
+
+
+def find_dist(
```

Comment:
This block of code to ensure pip is installed mover from the venv PEX_TOOL here: https://github.com/pantsbuild/pex/pull/2140/files#diff-44e8fd52cd3a63a2d3f2fc516418c2aa55e8ee96f6eb82b2186765a8b853922fL83-L180